Assets/MixedRealityToolkit/Utilities/MixedRealityToolkitSceneChecker.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
```

Comment:
Note: It would be nicer to have this utility class in the `Microsoft.MixedReality.Toolkit.Core.Utilities.Editor` assembly, which has an explicit Editor dependency. However, the `Microsoft.MixedReality.Toolkit` assembly in turn has a dependency on `Microsoft.MixedReality.Toolkit.Core.Utilities.Editor`, which would create a cycle. That should not be the case IMO, if possible we should try to remove editor code from the core assembly.

Reply:
Let's file an issue on that once this is in.

Reply:
There's a specific reason for that dependency, because the Mixed Reality Settings class needs to be compiled and ran before the core xrtk assemblies, or else some editor settings will not be properly initialized or setup as needed.

Now not all the classes in that assembly probably belong there, but things have been done for specific reasons and many people seem to be breaking and the effects are spreading though out much of the code base.

Reply:
What/where is the Mixed Reality Settings class?

Reply:
I've logged this here for future discussion: https://github.com/Microsoft/MixedRealityToolkit-Unity/issues/3952

Reply:
The MRTK core assembly should have no dependencies, in a perfect world. Let's identify the specific editor assembly classes and methods and investigate moving them into the core assembly, or at least factoring them into an isolated assembly that is safe for the core to reference (and avoid the cycle that @lukastonneMS mentions).